Remove code in common files only compiled for Apple's Windows port

brentfulgham opened the pull request:
#### a24ac189df77eb8e7d2d17f1b4bc8dc2386facc5
<pre>
Remove code in common files only compiled for Apple&apos;s Windows port
<a href="https://bugs.webkit.org/show_bug.cgi?id=252018">https://bugs.webkit.org/show_bug.cgi?id=252018</a>
&lt;rdar://problem/105243320&gt;

Reviewed by Alex Christensen.

This patch removes code that is no longer compiled on any platform, focused on the
CoreGraphics, CoreAnimation, and CFNetwork parts of the system.

It also moves some code from ImageSourceCG into UTIRegistry (since those methods were
only UTI-related).

* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/WebCorePrefix.h:
* Source/WebCore/loader/EmptyClients.cpp:
(WebCore::EmptyFrameLoaderClient::shouldCacheResponse): Deleted.
* Source/WebCore/loader/EmptyFrameLoaderClient.h:
* Source/WebCore/loader/FrameLoaderClient.h:
* Source/WebCore/loader/ResourceLoader.h:
* Source/WebCore/loader/cf/ResourceLoaderCFNet.cpp: Removed.
* Source/WebCore/platform/MIMETypeRegistry.cpp:
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
(WebCore::supportsAcceleratedFilterAnimations):
(WebCore::GraphicsLayer::supportsLayerType):
* Source/WebCore/platform/graphics/ca/PlatformCALayer.h:
* Source/WebCore/platform/graphics/cg/GraphicsContextCG.cpp:
(WebCore::setCGStyle):
(WebCore::GraphicsContextCG::didUpdateState):
* Source/WebCore/platform/graphics/cg/ImageBufferCGBackend.cpp:
(WebCore::ImageBufferCGBackend::contextColorSpace):
* Source/WebCore/platform/graphics/cg/ImageBufferUtilitiesCG.cpp:
* Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp:
(WebCore::ImageDecoderCG::setData):
(WebCore::sharedBufferGetBytesAtPosition): Deleted.
(WebCore::sharedBufferRelease): Deleted.
* Source/WebCore/platform/graphics/cg/ImageSourceCG.h: Removed.
* Source/WebCore/platform/graphics/cg/PDFDocumentImage.cpp:
(WebCore::PDFDocumentImage::drawPDFPage):
* Source/WebCore/platform/graphics/cg/PathCG.cpp:
(WebCore::Path::platformAddPathForRoundedRect):
* Source/WebCore/platform/graphics/cg/UTIRegistry.cpp:
(WebCore::MIMETypeForImageType):
(WebCore::preferredExtensionForImageType):
* Source/WebCore/platform/graphics/cg/UTIRegistry.h:
* Source/WebCore/platform/graphics/cocoa/MediaPlaybackTargetContext.h:
* Source/WebCore/platform/graphics/coretext/FontCascadeCoreText.cpp:
* Source/WebCore/platform/graphics/coretext/FontCoreText.cpp:
* Source/WebCore/platform/graphics/coretext/GlyphPageCoreText.cpp:
* Source/WebCore/platform/mediastream/cocoa/AudioMediaStreamTrackRendererInternalUnit.cpp:
* Source/WebCore/platform/network/NetworkStorageSession.h:
* Source/WebCore/platform/network/ProtectionSpace.h:
* Source/WebCore/platform/network/ProtectionSpaceBase.cpp:
* Source/WebCore/platform/network/ResourceHandle.h:
* Source/WebCore/platform/network/ResourceHandleClient.h:
* Source/WebCore/platform/network/ResourceHandleInternal.h:
(WebCore::ResourceHandleInternal::ResourceHandleInternal):
* Source/WebCore/platform/network/ResourceRequestBase.cpp:
* Source/WebCore/platform/network/cf/AuthenticationCF.cpp: Removed.
* Source/WebCore/platform/network/cf/AuthenticationCF.h: Removed.
* Source/WebCore/platform/network/cf/AuthenticationChallenge.h:
(WebCore::AuthenticationChallenge::nsURLAuthenticationChallenge const):
(WebCore::AuthenticationChallenge::cfURLAuthChallengeRef const): Deleted.
* Source/WebCore/platform/network/cf/CertificateInfoCFNet.cpp:
(WebCore::CertificateInfo::certificateChainFromSecTrust):
(WebCore::CertificateInfo::containsNonRootSHA1SignedCertificate const):
(WebCore::CertificateInfo::summary const):
* Source/WebCore/platform/network/cf/CookieStorageCFNet.cpp: Removed.
* Source/WebCore/platform/network/cf/CredentialStorageCFNet.cpp: Removed.
* Source/WebCore/platform/network/cf/LoaderRunLoopCF.cpp: Removed.
* Source/WebCore/platform/network/cf/LoaderRunLoopCF.h: Removed.
* Source/WebCore/platform/network/cf/NetworkStorageSessionCFNet.cpp:
(WebCore::NetworkStorageSession::cookieStorage const):
* Source/WebCore/platform/network/cf/ProtectionSpaceCFNet.cpp: Removed.
* Source/WebCore/platform/network/cf/ProtectionSpaceCFNet.h: Removed.
* Source/WebCore/platform/network/cf/ResourceError.h:
(WebCore::ResourceError::ResourceError):
* Source/WebCore/platform/network/cf/ResourceErrorCF.cpp: Removed.
* Source/WebCore/platform/network/cf/ResourceHandleCFNet.cpp: Removed.
* Source/WebCore/platform/network/cf/ResourceHandleCFURLConnectionDelegate.cpp: Removed.
* Source/WebCore/platform/network/cf/ResourceHandleCFURLConnectionDelegate.h: Removed.
* Source/WebCore/platform/network/cf/ResourceHandleCFURLConnectionDelegateWithOperationQueue.cpp: Removed.
* Source/WebCore/platform/network/cf/ResourceHandleCFURLConnectionDelegateWithOperationQueue.h: Removed.
* Source/WebCore/platform/network/cf/ResourceRequest.h:
(WebCore::ResourceRequest::ResourceRequest):
(WebCore::ResourceRequest::resourcePrioritiesEnabled):
* Source/WebCore/platform/network/cf/ResourceRequestCFNet.cpp:
(WebCore::ResourceRequest::isUserInitiatedKey):
(WebCore::setContentDispositionEncodingFallbackArray): Deleted.
(WebCore::copyContentDispositionEncodingFallbackArray): Deleted.
(WebCore::ResourceRequest::cfURLRequest const): Deleted.
(WebCore::setHeaderFields): Deleted.
(WebCore::toPlatformRequestCachePolicy): Deleted.
(WebCore::fromPlatformRequestCachePolicy): Deleted.
(WebCore::siteForCookies): Deleted.
(WebCore::ResourceRequest::doUpdatePlatformRequest): Deleted.
(WebCore::ResourceRequest::doUpdatePlatformHTTPBody): Deleted.
(WebCore::ResourceRequest::doUpdateResourceRequest): Deleted.
(WebCore::ResourceRequest::doUpdateResourceHTTPBody): Deleted.
(WebCore::ResourceRequest::setStorageSession): Deleted.
* Source/WebCore/platform/network/cf/ResourceRequestCFNet.h:
* Source/WebCore/platform/network/cf/ResourceResponse.h:
(WebCore::ResourceResponse::ResourceResponse):
* Source/WebCore/platform/network/cf/ResourceResponseCFNet.cpp: Removed.
* Source/WebCore/platform/network/cf/SocketStreamHandleImplCFNet.cpp:
(WebCore::SocketStreamHandleImpl::SocketStreamHandleImpl):
(WebCore::SocketStreamHandleImpl::createStreams):
* Source/WebCore/platform/network/cf/SynchronousLoaderClientCFNet.cpp: Removed.
* Source/WebCore/platform/network/cocoa/CredentialCocoa.h:
* Source/WebCore/platform/network/cocoa/ProtectionSpaceCocoa.h:
* Source/WebCore/platform/network/mac/ResourceErrorMac.mm:
(WebCore::ResourceError::ResourceError):
* Source/WebCore/platform/network/mac/ResourceHandleMac.mm:
* Source/WebCore/platform/win/TemporaryLinkStubs.cpp:
(WebCore::stopObservingCookieChanges):
* Source/WebKitLegacy/WebCoreSupport/NetworkStorageSessionMap.cpp:
(NetworkStorageSessionMap::switchToNewTestingSession):
(NetworkStorageSessionMap::ensureSession):

Canonical link: <a href="https://commits.webkit.org/260136@main">https://commits.webkit.org/260136@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d6ac474e962f34143e6f5fb86484bb328b4bf9e3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107181 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16227 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/39966 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116353 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/115827 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111082 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/17685 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/7529 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99358 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/112946 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/13364 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96423 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/40968 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95264 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28059 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/82733 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/9308 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29400 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/9921 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/6395 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/15469 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48979 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7008 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11464 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/platform/network/ResourceHandleInternal.h:
```diff
@@ -108,20 +98,11 @@ class ResourceHandleInternal {
-#endif
-#if PLATFORM(COCOA) || USE(CFURLCONNECTION)
     RetainPtr<CFURLStorageSessionRef> m_storageSession;
+    bool m_startWhenScheduled { false };
```
nit: putting this with other bools would pack better.

brentfulgham replied:
Oh! I will do!

brentfulgham committed:
Remove code in common files only compiled for Apple's Windows port
https://bugs.webkit.org/show_bug.cgi?id=252018
<rdar://problem/105243320>

Reviewed by Alex Christensen.

This patch removes code that is no longer compiled on any platform, focused on the
CoreGraphics, CoreAnimation, and CFNetwork parts of the system.

It also moves some code from ImageSourceCG into UTIRegistry (since those methods were
only UTI-related).

* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/WebCorePrefix.h:
* Source/WebCore/loader/EmptyClients.cpp:
(WebCore::EmptyFrameLoaderClient::shouldCacheResponse): Deleted.
* Source/WebCore/loader/EmptyFrameLoaderClient.h:
* Source/WebCore/loader/FrameLoaderClient.h:
* Source/WebCore/loader/ResourceLoader.h:
* Source/WebCore/loader/cf/ResourceLoaderCFNet.cpp: Removed.
* Source/WebCore/platform/MIMETypeRegistry.cpp:
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
(WebCore::supportsAcceleratedFilterAnimations):
(WebCore::GraphicsLayer::supportsLayerType):
* Source/WebCore/platform/graphics/ca/PlatformCALayer.h:
* Source/WebCore/platform/graphics/cg/GraphicsContextCG.cpp:
(WebCore::setCGStyle):
(WebCore::GraphicsContextCG::didUpdateState):
* Source/WebCore/platform/graphics/cg/ImageBufferCGBackend.cpp:
(WebCore::ImageBufferCGBackend::contextColorSpace):
* Source/WebCore/platform/graphics/cg/ImageBufferUtilitiesCG.cpp:
* Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp:
(WebCore::ImageDecoderCG::setData):
(WebCore::sharedBufferGetBytesAtPosition): Deleted.
(WebCore::sharedBufferRelease): Deleted.
* Source/WebCore/platform/graphics/cg/ImageSourceCG.h: Removed.
* Source/WebCore/platform/graphics/cg/PDFDocumentImage.cpp:
(WebCore::PDFDocumentImage::drawPDFPage):
* Source/WebCore/platform/graphics/cg/PathCG.cpp:
(WebCore::Path::platformAddPathForRoundedRect):
* Source/WebCore/platform/graphics/cg/UTIRegistry.cpp:
(WebCore::MIMETypeForImageType):
(WebCore::preferredExtensionForImageType):
* Source/WebCore/platform/graphics/cg/UTIRegistry.h:
* Source/WebCore/platform/graphics/cocoa/MediaPlaybackTargetContext.h:
* Source/WebCore/platform/graphics/coretext/FontCascadeCoreText.cpp:
* Source/WebCore/platform/graphics/coretext/FontCoreText.cpp:
* Source/WebCore/platform/graphics/coretext/GlyphPageCoreText.cpp:
* Source/WebCore/platform/mediastream/cocoa/AudioMediaStreamTrackRendererInternalUnit.cpp:
* Source/WebCore/platform/network/NetworkStorageSession.h:
* Source/WebCore/platform/network/ProtectionSpace.h:
* Source/WebCore/platform/network/ProtectionSpaceBase.cpp:
* Source/WebCore/platform/network/ResourceHandle.h:
* Source/WebCore/platform/network/ResourceHandleClient.h:
* Source/WebCore/platform/network/ResourceHandleInternal.h:
(WebCore::ResourceHandleInternal::ResourceHandleInternal):
* Source/WebCore/platform/network/ResourceRequestBase.cpp:
* Source/WebCore/platform/network/cf/AuthenticationCF.cpp: Removed.
* Source/WebCore/platform/network/cf/AuthenticationCF.h: Removed.
* Source/WebCore/platform/network/cf/AuthenticationChallenge.h:
(WebCore::AuthenticationChallenge::nsURLAuthenticationChallenge const):
(WebCore::AuthenticationChallenge::cfURLAuthChallengeRef const): Deleted.
* Source/WebCore/platform/network/cf/CertificateInfoCFNet.cpp:
(WebCore::CertificateInfo::certificateChainFromSecTrust):
(WebCore::CertificateInfo::containsNonRootSHA1SignedCertificate const):
(WebCore::CertificateInfo::summary const):
* Source/WebCore/platform/network/cf/CookieStorageCFNet.cpp: Removed.
* Source/WebCore/platform/network/cf/CredentialStorageCFNet.cpp: Removed.
* Source/WebCore/platform/network/cf/LoaderRunLoopCF.cpp: Removed.
* Source/WebCore/platform/network/cf/LoaderRunLoopCF.h: Removed.
* Source/WebCore/platform/network/cf/NetworkStorageSessionCFNet.cpp:
(WebCore::NetworkStorageSession::cookieStorage const):
* Source/WebCore/platform/network/cf/ProtectionSpaceCFNet.cpp: Removed.
* Source/WebCore/platform/network/cf/ProtectionSpaceCFNet.h: Removed.
* Source/WebCore/platform/network/cf/ResourceError.h:
(WebCore::ResourceError::ResourceError):
* Source/WebCore/platform/network/cf/ResourceErrorCF.cpp: Removed.
* Source/WebCore/platform/network/cf/ResourceHandleCFNet.cpp: Removed.
* Source/WebCore/platform/network/cf/ResourceHandleCFURLConnectionDelegate.cpp: Removed.
* Source/WebCore/platform/network/cf/ResourceHandleCFURLConnectionDelegate.h: Removed.
* Source/WebCore/platform/network/cf/ResourceHandleCFURLConnectionDelegateWithOperationQueue.cpp: Removed.
* Source/WebCore/platform/network/cf/ResourceHandleCFURLConnectionDelegateWithOperationQueue.h: Removed.
* Source/WebCore/platform/network/cf/ResourceRequest.h:
(WebCore::ResourceRequest::ResourceRequest):
(WebCore::ResourceRequest::resourcePrioritiesEnabled):
* Source/WebCore/platform/network/cf/ResourceRequestCFNet.cpp:
(WebCore::ResourceRequest::isUserInitiatedKey):
(WebCore::setContentDispositionEncodingFallbackArray): Deleted.
(WebCore::copyContentDispositionEncodingFallbackArray): Deleted.
(WebCore::ResourceRequest::cfURLRequest const): Deleted.
(WebCore::setHeaderFields): Deleted.
(WebCore::toPlatformRequestCachePolicy): Deleted.
(WebCore::fromPlatformRequestCachePolicy): Deleted.
(WebCore::siteForCookies): Deleted.
(WebCore::ResourceRequest::doUpdatePlatformRequest): Deleted.
(WebCore::ResourceRequest::doUpdatePlatformHTTPBody): Deleted.
(WebCore::ResourceRequest::doUpdateResourceRequest): Deleted.
(WebCore::ResourceRequest::doUpdateResourceHTTPBody): Deleted.
(WebCore::ResourceRequest::setStorageSession): Deleted.
* Source/WebCore/platform/network/cf/ResourceRequestCFNet.h:
* Source/WebCore/platform/network/cf/ResourceResponse.h:
(WebCore::ResourceResponse::ResourceResponse):
* Source/WebCore/platform/network/cf/ResourceResponseCFNet.cpp: Removed.
* Source/WebCore/platform/network/cf/SocketStreamHandleImplCFNet.cpp:
(WebCore::SocketStreamHandleImpl::SocketStreamHandleImpl):
(WebCore::SocketStreamHandleImpl::createStreams):
* Source/WebCore/platform/network/cf/SynchronousLoaderClientCFNet.cpp: Removed.
* Source/WebCore/platform/network/cocoa/CredentialCocoa.h:
* Source/WebCore/platform/network/cocoa/ProtectionSpaceCocoa.h:
* Source/WebCore/platform/network/mac/ResourceErrorMac.mm:
(WebCore::ResourceError::ResourceError):
* Source/WebCore/platform/network/mac/ResourceHandleMac.mm:
* Source/WebCore/platform/win/TemporaryLinkStubs.cpp:
(WebCore::stopObservingCookieChanges):
* Source/WebKitLegacy/WebCoreSupport/NetworkStorageSessionMap.cpp:
(NetworkStorageSessionMap::switchToNewTestingSession):
(NetworkStorageSessionMap::ensureSession):

Canonical link: https://commits.webkit.org/260136@main
